### PR TITLE
quote component

### PR DESCRIPTION
**Pull request by monikajoon**

### CODE REVIEW

File: lib/components/molecules/Quote/Quote.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  className: string,
+  content: string,
+  contentSrc: string,
+  author?: string,
+  company?: string,
+};
+
+const Quote = ({ content, contentSrc, author, company, className, ...otherProps }: Props): Node => {
+  return (
+    <section {...otherProps} className={classnames('quote-wrap', className)}>
```
Comment (vinodloha) [inline lib/components/molecules/Quote/Quote.js]: Why does your Quote component makes use of a section?

Comment (monikajoon) [reply lib/components/molecules/Quote/Quote.js]: fixed.

File: lib/components/molecules/Quote/Quote.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  author?: string,
+  company?: string,
+};
+
+const Quote = ({ content, contentSrc, author, company, className, ...otherProps }: Props): Node => {
+  return (
+    <div {...otherProps} className={classnames('quote-wrap', className)}>
+      <div className="quote__mark">&rdquo;</div>
+      <blockquote className="quote__content" cite={contentSrc}>
+        <p>{content}</p>
```
Comment (ankurkakroo2) [inline lib/components/molecules/Quote/Quote.js]: We should make / accept the content as "children"

File: lib/components/molecules/Quote/Quote.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        <footer className="quote__author">
+          {author !== '' ? <cite>{author}</cite> : ''}
```
Comment (ankurkakroo2) [inline lib/components/molecules/Quote/Quote.js]: We should accept classes to style the citation section.
Accept citationClassNames

File: lib/components/molecules/Quote/Quote.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+          {author !== '' && company !== '' ? <span>, </span> : ''}
```
Comment (ankurkakroo2) [inline lib/components/molecules/Quote/Quote.js]: accept a prop named authorSeparator which will be defaulted to a comma (,)

File: lib/components/molecules/Quote/Quote.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  content: string,
+  contentSrc: string,
+  author?: string,
+  company?: string,
+};
+
+const Quote = ({ content, contentSrc, author, company, className, ...otherProps }: Props): Node => {
+  return (
+    <div {...otherProps} className={classnames('quote-wrap', className)}>
+      <div className="quote__mark">&rdquo;</div>
```
Comment (ankurkakroo2) [inline lib/components/molecules/Quote/Quote.js]: May be a flag here displayQuoteMarks as a boolean
or a prop named quoteIndicator which will default to "

File: lib/components/molecules/Quote/Quote.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  className: string,
+  content: string,
+  contentSrc: string,
+  author?: string,
+  company?: string,
+};
+
+const Quote = ({ content, contentSrc, author, company, className, ...otherProps }: Props): Node => {
+  return (
+    <div {...otherProps} className={classnames('quote-wrap', className)}>
```
Comment (vinodloha) [inline lib/components/molecules/Quote/Quote.js]: Make the block-quote outside most component. Examples below 

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/blockquote 
http://html5doctor.com/blockquote-q-cite/

File: lib/components/molecules/Quote/Quote.style.js [hunk 7]
```diff
@@ -0,0 +1,22 @@
+import { css } from 'styled-components';
+
+const styles = css`
+  .quote {
+    &__mark {
+      font-size: 60px;
```
Comment (ankurkakroo2) [inline lib/components/molecules/Quote/Quote.style.js]: dont we have variable for standard font attributes / variables to use from ?

File: lib/components/molecules/Quote/Quote.style.js [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+      float: left;
```
Comment (ankurkakroo2) [inline lib/components/molecules/Quote/Quote.style.js]: Lets refrain from using floats to control the layout

File: lib/components/molecules/Quote/Quote.style.js [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+    }
+    &__content {
+      padding-top: 5px;
```
Comment (ankurkakroo2) [inline lib/components/molecules/Quote/Quote.style.js]: this should be driven from variables and should be in multiples of a standard multiplier

File: lib/components/molecules/Quote/Quote.style.js [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+      ${props => (props.contentStyle ? props.contentStyle : '')};
+    }
+
+    &__author {
```
Comment (ankurkakroo2) [inline lib/components/molecules/Quote/Quote.style.js:16]: Lets use flex to control the layout

**Comment by monikajoon** [comment]
@vinodloha , @ankurkakroo2 , Have fixed the review points. Please have a look.

File: lib/components/molecules/Quote/Quote.js [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+        {company !== '' ? <cite>{company}</cite> : ''}
+        {author !== '' && company !== '' ? (
+          <span>
+            {authorSeparator}
+            &nbsp;
+          </span>
+        ) : (
+          ''
+        )}
+        {author !== '' ? <cite>{author}</cite> : ''}
```
Comment (vinodloha) [inline lib/components/molecules/Quote/Quote.js:42]: {author !== '' ? <cite>{author}</cite> : ''} . can be written as {author && <cite>{author}</cite>}

Change rest of the settings accordingly... Ideally author in quote componnet should not be optional and have a default value as "anonymous"

File: lib/components/molecules/Quote/Quote.js [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+  author,
+  company,
+  className,
+  authorSeparator,
+  quoteIndicator,
+  ...otherProps
+}: Props): Node => {
+  return (
+    <blockquote {...otherProps} cite={contentSrc} className={classnames('quote-wrap', className)}>
+      <div className={classnames('quote__mark', className)}>{quoteIndicator}</div>
```
Comment (vinodloha) [inline lib/components/molecules/Quote/Quote.js:30]: Why the quoteIndicator can't be done with CSS ::before or ::after?

File: lib/components/molecules/Quote/Quote.js [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+      <div className={classnames('quote__content', className)}>{children}</div>
+      <footer className={classnames('quote__author', className)}>
+        {company !== '' ? <cite>{company}</cite> : ''}
+        {author !== '' && company !== '' ? (
```
Comment (vinodloha) [inline lib/components/molecules/Quote/Quote.js:34]: Unnecessary condition, if you put this separator with  

{company && <cite>{company} {authorSeparator}</cite> }

File: lib/components/molecules/Quote/Quote.style.js [hunk 14]
```diff
@@ -0,0 +1,25 @@
+import { css } from 'styled-components';
+
+const styles = css`
+  display: flex;
```
Comment (vinodloha) [inline lib/components/molecules/Quote/Quote.style.js:4]: Don't see the use of Flex here in this context.

File: lib/components/molecules/Quote/Quote.style.js [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+  flex-flow: row wrap;
+  .quote {
+    &__mark {
+      flex: 1 5%;
+      ${props => (props.markStyle ? props.markStyle : '')};
+    }
```
Comment (vinodloha) [inline lib/components/molecules/Quote/Quote.style.js:10]: Could be done through positioning and content: ' " ';

File: lib/components/molecules/Quote/Quote.style.js (hunk 10 above)
Comment (vinodloha) [reply lib/components/molecules/Quote/Quote.style.js:16]: But here in this component there is no layout to control. :D

File: lib/components/molecules/Quote/tests/Quote.style.test.js [hunk 16]
```diff
@@ -0,0 +1,16 @@
...
+import 'jest-styled-components';
+
+import Quote from '../Quote';
+import Theme from '../../../../styles/theme';
+
+describe('Quote', () => {
+  let quoteComp = '';
+  const theme = { ...Theme };
+
+  test('should render quote correctly', () => {
```
Comment (vinodloha) [inline lib/components/molecules/Quote/tests/Quote.style.test.js:12]: This test is not enough.

**Review by vinodloha — CHANGES_REQUESTED** [review]
The component looks over engineered. Please fix review comments.

**Comment by vinodloha** [comment]
The feature is already implemented and merged

**Pull request closed without merging**